### PR TITLE
2.x: Automatically publish the generated JavaDocs from CI

### PR DESCRIPTION
**Pull request by akarnokd**
This PR adds a script that publishes the generated JavaDocs from CI back to the `gh-pages` branch upon successful snapshot and release builds (this latter needs testing though via -RC releases).

The snapshot JavaDocs will go into the same directory: http://reactivex.io/RxJava/2.x/javadoc/snapshot/

The release builds update
- http://reactivex.io/RxJava/javadoc/ while preserving the old `rx` links,
- http://reactivex.io/RxJava/2.x/javadoc/ for the latest 2.x and
- http://reactivex.io/RxJava/2.x/javadoc/2.1.13/ for the specific release version.

The `push.sh` script uses a Travis-CI secured token `GITHUB_TOKEN` to talk to the GitHub API. This token was created for my account via https://github.com/settings/tokens (requires repo direct push access), and provided to Travis via "Environment variables" on the page https://travis-ci.org/ReactiveX/RxJava/settings .

If, for some reason the publishing fails, the build still counts as successful. The reason for this is that restarting the otherwise successful build will try to reupload the same library version into maven which fails. In this case, the JavaDocs can be updated via the old manual method.

### CODE REVIEW

**Comment by akarnokd** [comment]
PRs don't trigger: https://travis-ci.org/ReactiveX/RxJava/builds/375403314#L4010

Branches do trigger: https://travis-ci.org/ReactiveX/RxJava/builds/375403308#L4018

File: .travis.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@ script: gradle/buildViaTravis.sh
 # Code coverage
 after_success:
   - bash <(curl -s https://codecov.io/bash)
+  - bash gradle/push.sh
```
Comment (vanniktech) [inline .travis.yml]: push_javadoc?

File: gradle/push.sh [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+git add *.js
+git add *package-list*
+
+# remove tracked but deleted files
+echo -e "Removing deleted files"
+git add -u
+
+# commit all
+echo -e "commit Travis build: $TRAVIS_BUILD_NUMBER for $buildTag"
+git commit --message "Travis build: $TRAVIS_BUILD_NUMBER for $buildTag"
```
Comment (vanniktech) [inline gradle/push.sh]: Should we include in the message that I'll update the documentation?

**Review by vanniktech — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (akarnokd) [reply .travis.yml]: Okay.

File: gradle/push.sh (hunk 2 above)
Comment (akarnokd) [reply gradle/push.sh]: I don't think there is much ambiguity. These go into the `gh-pages` tab: https://github.com/ReactiveX/RxJava/tree/gh-pages

**Review by vanniktech — APPROVED** [review]

**Comment by akarnokd** [comment]
Looks like the release mode also works:

- http://reactivex.io/RxJava/javadoc/
- http://reactivex.io/RxJava/2.x/javadoc/
- http://reactivex.io/RxJava/2.x/javadoc/2.1.14-RC1/

**Pull request merged**